.gitattributes

```diff
@@ -1,3 +1,3 @@
 # Do not add git-lfs attributes to this file in stable. It causes problems with Cocoapods.
```

Comment:
should we remove this comment? is it accurate?

Reply:
As it stands, the comment appears to disagree with the change proposed in this PR. Should `.gitattributes merge=gitattributes` be uncommented in the stable branch?

Reply:
It should be uncommented - it ensures that we don't attempt to merge any LFS-tracked files.  The purpose of the custom merge driver is to ensure that we perform a no-op on the tracked files.  

What makes you think that the comment in the PR is at odds with restoring the no-op driver?

Reply:
"Do not add git-lfs attributes to this file in stable. It causes problems with Cocoapods." reads to me as "we should not be including `.gitattributes merge=gitattributes` as an uncommented line in this file in stable.

Reply:
@yarneo The comment on the first line is accurate. Requiring Git LFS support in `stable` means that it becomes required for the `stable` branch. That was the reason for the introduction of the custom driver and the different `.gitattributes` in the two branches.  The original conversation and relevant PRs are linked to #5956.

Reply:
@jverkoey @yarneo If we believe the underlying issues have changed, it would be worth documenting them. I did not see such explanation/documentation in https://github.com/material-components/material-components-ios/pull/8221.

Reply:
@jverkoey That comment was added at the same time the custom merge strategy was added. It was intended as a warning that no changes should take place to the file in the future. In 1bd980482431194c00508b5c9d332db064f6cb4d we dropped the protection against merging from `develop`.  I'm not sure why the warning was not heeded at that time, but I'm attempting to restore the protection.  If you believe we should now require Git LFS support for the `stable` branch, let's please update our documentation and processes and we can close this PR.

Reply:
As an artifact of the current version in `stable` here's what happens when someone attempts to clone our repository without Git LFS installed:

```
rsmoore-macbookpro3:tmp rsmoore$ git clone git@github.com:material-components/material-components-ios.git
Cloning into 'material-components-ios'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (18/18), done.
remote: Total 72036 (delta 4), reused 15 (delta 4), pack-reused 72012
Receiving objects: 100% (72036/72036), 144.37 MiB | 12.01 MiB/s, done.
Resolving deltas: 100% (49534/49534), done.
git-lfs filter-process: git-lfs: command not found
fatal: the remote end hung up unexpectedly
warning: Clone succeeded, but checkout failed.
You can inspect what was checked out with 'git status'
and retry the checkout with 'git checkout -f HEAD'
```

Reply:
> That was the reason for the introduction of the custom driver and the different .gitattributes in the two branches.

What is a custom driver?

> If you believe we should now require Git LFS support for the stable branch

I do not; I want us to not regress on the issue that was discovered in #6350. Given that we apparently have regressed, I'm not convinced that the documentation here or elsewhere is proving to be sufficiently guarding us against regressing.

What can we do to make sure we don't regress on this again?

Importantly, at least for me, I do not understand what "Do not add git-lfs attributes to this file in stable. It causes problems with Cocoapods." means. Uncommenting `.gitattributes merge=gitattributes` looks like we are adding some git-lfs attributes to this file.

Reply:
@jverkoey I must defer to the [`.gitattributes`](https://git-scm.com/docs/gitattributes) documentation for what a filter driver is. I think it has the best description.

Reply:
Updated the comments in the file to be more explicit and guard against these accidents in the future.  Please take a look and let me know what you think.